### PR TITLE
Avoid duplication in `toggle-files-button`

### PR DESCRIPTION
**Pull request by fregante**
- Closes https://github.com/refined-github/refined-github/issues/5692
- Closes https://github.com/refined-github/refined-github/issues/5693

URL: https://github.com/refined-github/refined-github

https://user-images.githubusercontent.com/1402241/174477063-52f36daa-e552-4552-ada6-a51fffe5718a.mov

### CODE REVIEW

File: source/helpers/attach-element.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	position: Position;
+	allowMissingAnchor?: boolean;
+}
+
+// Get a unique position in the code
+function getSnapshotUUID(index = 3): string {
+	return hashString(new Error('Get stack').stack!.split('\n')[index]);
+}
+
+export default function attach<NewElement extends Element>({
```
Comment (yakov116) [inline source/helpers/attach-element.ts:21]: Can we call this function `attachElement`?

File: source/helpers/attach-element.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import select from 'select-dom';
+
```
Comment (yakov116) [inline source/helpers/attach-element.ts:2]:
```suggestion
```

File: source/helpers/attach-element.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	// eslint-disable-next-line @typescript-eslint/ban-types --  Allows dom traversing without requiring `!`
+	anchor: Element | string | undefined | null;
+	getNewElement: () => NewElement;
+	className?: string;
+	position: Position;
+	allowMissingAnchor?: boolean;
+}
+
+// Get a unique position in the code
+function getSnapshotUUID(index = 3): string {
```
Comment (yakov116) [inline source/helpers/attach-element.ts:17]: I am not 100% sure what you are doing here. But can we use https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID ?

Comment (fregante) [reply source/helpers/attach-element.ts:17]: It's not random, it's an automatic identifier/key of the call. It's the same every time it's called (even across sessions), but it's different for every caller.

The index defines which call point is considered (from the stack).

Maybe it should just be named `callerID` 🙂 


This lets us avoid custom classes unless necessary (in this case they're necessary though)

Comment (yakov116) [reply source/helpers/attach-element.ts:17]: This was a really good lesson something I never knew.
Thanks

**Pull request merged**